### ISSUE TITLE
Button: don't set aria-label attribute if label (accessible string) is already set

### ISSUE THREAD

**Issue by brightpixels** [Capgemini/dcx-react-library#334]
**Describe the bug**
https://github.com/Capgemini/dcx-react-library/blob/main/src/button/Button.tsx#L120

As you can see above, if ariaLabel prop isn't passed in by the developer, the fallback for aria label is `${type}-button`. In many cases, this becomes "submit-button". 

Aria label shouldn't be set when there is accessible string (in this case label). Aria label should be conditional.
`<button {...(label ? {ariaLabel: ariaLabel} : {})}>...</button>`

When aria label is being set, it creates issues when testing the UI with @testing-library/cypress library (specifically using `cy.findByRole`) because it gives higher importance to aria label attribute than the text/label of the button itself. In pages, with multiple buttons, all the buttons carry the same aria label, unless we explicity pass in aria label to each button (even though we are setting a label already).

**To Reproduce**
1. Use a button from the DCX library and don't specify a ariaLabel props
2. Use Cypress and helper library @testing-library/cypress library and find the element using `cy.findByRole()`

**Expected behaviour**
- IF a label is passed in, don't set aria-label to a generic default.
So something like this on line 120
`<button {...(label ? {ariaLabel: ariaLabel} : {})}>...</button>`

Please follow the following instruction to create the new branch:
```bash
git checkout release/0.6
git pull
yarn
git checkout -b 'feature/button-aria'
```

**Comment by cg-julian-taylor** [comment]
I may be overthinking but is there ever a use case where we would pass in both a label and an aria-label?

**Comment by brightpixels** [comment]
No. If label is passed in, then we should never use aria-label. If both are passed in, then only set the label. 

Don't see issue with the default aria-labels listed above.

**Comment by daniele-zurico** [comment]
PR merged
